File: binarysearch.py
# Binary search
# Only for sorted array

arr = [1, 2, 2, 3, 5, 6, 8, 9]

# Binary search example
target = 6
# Initial index, can use for while loop
a = 0
b = len(arr) - 1

# while method
while a <= b:
    k = int((a + b)/2)
    if arr[k] == target:
        print(k)
    if arr[k] > target:
        b = k - 1
    else:
        a = k + 1

File: binarysearch.cpp
// Binary search using for loop
// Only for sorted array

#include <iostream>
using namespace std;

#define ll long long

int main() {
    int arr[] = {1, 2, 2, 3, 5, 6, 8, 9};
    int n = sizeof(arr) / sizeof(*arr);
    int k = 0;
    int target = 6;
    
    for (int b = n/2; b >= 1; b /= 2)
    {
        while (k + b < n && arr[k+b] <= target)
        {
            k += b;
        }
        if (arr[k] == target)
        {
            cout << k << "\n";
        }
    }

    return 0;
}
